components/mediation/data-publishers/org.wso2.micro.integrator.observability/src/main/java/org/wso2/micro/integrator/observability/handler/metrichandler/MetricHandler.java

```diff
@@ -0,0 +1,440 @@
+import org.apache.synapse.SynapseConstants;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.rest.API;
+import org.apache.synapse.rest.RESTConstants;
+import org.apache.synapse.rest.RESTUtils;
+import org.apache.synapse.transport.nhttp.NhttpConstants;
+import org.wso2.config.mapper.ConfigParser;
+import org.wso2.micro.integrator.core.internal.MicroIntegratorBaseConstants;
+import org.wso2.micro.integrator.observability.handler.metrichandler.prometheus.reporter.PrometheusReporter;
+import org.wso2.micro.integrator.observability.handler.util.MetricConstants;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * Class for instrumenting Metrics by wrapping the implementation and
+ * there by allows the implementation to plug into the handler in runtime.
+ */
+public class MetricHandler extends AbstractExtendedSynapseHandler {
+    private static final String METRIC_REPORTER = "metric_reporter";
+    private static final String DELIMITER = "/";
+    private static final String EMPTY = "";
+
+    private MetricReporter metricReporterInstance;
+    private Class loadedMetricClass;
+    private static Log log = LogFactory.getLog(MetricHandler.class);
+
+    private static final String serverPortOffset = System.getProperty(MetricConstants.PORT_OFFSET);
+    private static final String host = System.getProperty(MicroIntegratorBaseConstants.LOCAL_IP_ADDRESS);
+    private static final String port = System.getProperty(MetricConstants.HTTP_PORT);
+    private static final String javaVersion = System.getProperty(MetricConstants.JAVA_VERSION);
+    private static final String javaHome = System.getProperty(MetricConstants.JAVA_HOME);
+
+    @Override
+    public boolean handleServerInit() {
+        getMetricReporter();
+        metricReporterInstance.initMetrics();
+
+        metricReporterInstance.serverUp(host, port, javaHome, javaVersion);
+        return true;
+    }
+
+    /**
+     * Load the MetricReporter class from the deployment.toml file if a user has defined a MetricReporter.
+     * Use default PrometheusReporter if the user hasn't defined a MetricReporter or an error occurs
+     * during custom MetricReporter class invocation.
+     */
+    private void getMetricReporter() {
+        Map<String, Object> configs = ConfigParser.getParsedConfigs();
+        Object metricReporterClass = configs.get(MetricConstants.METRIC_HANDLER + "." + METRIC_REPORTER);
+
+        if (metricReporterClass != null) {
+            try {
+                loadedMetricClass = Class.forName(metricReporterClass.toString());
+                metricReporterInstance = (MetricReporter) loadedMetricClass.newInstance();
+                log.debug("The class " + metricReporterClass + "loaded successfully");
+            } catch (IllegalAccessException | ClassNotFoundException | InstantiationException e) {
+                log.error("Error in loading the class" + metricReporterClass.toString() +
```

Comment:
Add spaces after "class" and before "Hence"

Reply:
Fixed

Reply:
I still see that the spaces are missing. Can we fix it?

Reply:
Fixed